metagraph/src/annotation/taxonomy/tax_classifier.hpp

```diff
@@ -0,0 +1,192 @@
+     * @param [modified] 'node_scores' -> the current score for each node in the tree.
+     * @param [modified] 'nodes_already_propagated' -> the set of nodes that were previously processed.
+     * @param [modified] 'best_lca' -> the current classification prediction (node that exceeds the `desired_number_kmers` threshold and is placed as close as possible to the leaves).
+     * @param [modified] 'best_lca_dist_to_root' -> the distance to the root for the current classification prediction.
+     */
+    void update_scores_and_lca(const TaxId start_node,
+                               const tsl::hopscotch_map<TaxId, uint64_t> &num_kmers_per_node,
+                               const uint64_t desired_number_kmers,
+                               tsl::hopscotch_map<TaxId, uint64_t> *node_scores,
+                               tsl::hopscotch_set<TaxId> *nodes_already_propagated,
+                               TaxId *best_lca,
+                               uint32_t *best_lca_dist_to_root) const;
+
+    /**
+     * Get the list of LCA taxid for each kmer in a given sequences.
+     * The sequence can be given in forward or in reversed orientation.
+     */
+    virtual std::vector<TaxId> get_lca_taxids_for_seq(const std::string_view &sequence, bool reversed) const = 0;
+
+    LabelType label_type;
+
+    /**
+     * node_depth returns the depth for each node in the taxonomic tree.
+     * The root is the unique node with maximal depth and all the leaves have depth equal to 1.
+     */
+    tsl::hopscotch_map<TaxId, uint32_t> node_depth;
+
+    TaxId root_node;
+
+    /**
+     *  node_parent stores a taxonomic tree representation as a taxid to taxid parent list.
+     */
+    tsl::hopscotch_map<TaxId, TaxId> node_parent;
+
+    tsl::hopscotch_map<std::string, TaxId> accversion_to_taxid_map;
+
+    double _lca_coverage_rate;
+    double _kmers_discovery_rate;
+};
+
+class TaxonomyClsImportDB : public TaxonomyBase {
+  public:
+    // todo implement
+    TaxonomyClsImportDB(const std::string &taxdb_filepath,
+                        const double lca_coverage_rate,
+                        const double kmers_discovery_rate);
+
+  private:
+    std::vector<TaxId> get_lca_taxids_for_seq(const std::string_view &sequence, bool reversed) const;
+    TaxId find_lca(const std::vector<TaxId> &taxids) const;
+};
+
+class TaxonomyClsAnno : public TaxonomyBase {
+  public:
+    /**
+     * TaxonomyCls constructor
+     *
+     * @param [input] anno -> the annotation matrix
+     * @param [input] lca_coverage_rate -> threshold used for taxonomic classification.
+     * @param [input] kmers_discovery_rate -> threshold used for taxonomic classification.
+     * @param [input] tax_tree_filepath ->  path to a taxonomic tree ("nodes.dmp" file).
+     * @param [input] label_taxid_map_filepath ->  path to acccession version to taxid lookup table (".accession2taxid" file). Mandatory if the label doesn't contain the 'taxid'.
```

Comment:
the line is too long

Reply:
I've split it into multiple lines. I was not considering the comments, but now I fixed this in the entire PR